lib/validation-stream.js

```diff
@@ -9,6 +9,12 @@ var Duplex = require('readable-stream').Duplex;
 
 var XmlParser = require('./xml-parser');
 
+/** @arg {?string} str Input string to check.
+    @return {!boolean} True if input is undefined, null, or entirely whitespace. */
+function isBlank(str) {
```

Comment:
Although I didn't unit test this function, I prefer many small functions for unit testing as well as composability and because I don't want to think about more than I have to at any time. One of the downsides is that the whole is harder to see.